src/OpenLoco/Ui/WindowManager.cpp

```diff
@@ -794,11 +794,44 @@ namespace OpenLoco::Ui::WindowManager
     // TODO: hook
     Window* bringToFront(Window* w)
     {
-        registers regs;
-        regs.esi = (uint32_t)w;
-        call(0x004CC750, regs);
+        if (w->flags & (WindowFlags::stick_to_back | WindowFlags::stick_to_front))
+        {
+            return w;
+        }
+
+        Window* frontMostWnd = nullptr;
+        for (auto i = count(); i != 0; --i)
+        {
+            if (!(_windows[i - 1].flags & WindowFlags::stick_to_back))
+            {
+                frontMostWnd = &_windows[i - 1];
+                break;
+            }
+        }
+        if (frontMostWnd != nullptr && frontMostWnd != &_windows[0] && frontMostWnd != w)
+        {
+            std::swap(*frontMostWnd, *w);
+            w = frontMostWnd;
+            w->invalidate();
+        }
+
+        auto right = w->x + w->width;
+        // If window is almost offscreen to the right
+        if (right < 20)
+        {
+            right = 20 - w->x;
```

Comment:
At this point `right` is the amount of pixels the window moved horizontally. Should be a new variable.